### PR TITLE
fix: Correct the order of generating default and overridden broker capacities in Capacity.processCapacityEntries

### PR DESCRIPTION
**Pull request by Xander-run**
### Type of change

- Bugfix

### Description

Correct the order of generating default and overridden broker capacities in Capacity.processCapacityEntries. The order after the change will be:

1. Initialize the broker capacities based on `spec.cruiseControl.brokerCapacity`.
2. Override the default capacities base on `spec.cruiseControl.brokerCapacity.overrides`.

This reduces the unneeded loop when generating broker capacity configurations and will resolve issue https://github.com/strimzi/strimzi-kafka-operator/issues/10465 

### Checklist

- [ ] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

I ran and passed test class `io.strimzi.operator.cluster.model.CruiseControlTest` for this change.

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java [hunk 1]
```diff
@@ -365,44 +365,45 @@ private void processCapacityEntries(CruiseControlSpec spec, Set<NodeRef> kafkaBr
...
-                                            brokerCapacityEntry.setCpu(processCpu(override, brokerCapacity, kafkaBrokerResources.get(Integer.toString(id))));
-                                            brokerCapacityEntry.setInboundNetwork(inboundNetwork);
-                                            brokerCapacityEntry.setOutboundNetwork(outboundNetwork);
-                                        } else {
-                                            LOGGER.warnCr(reconciliation, "Duplicate broker id {} found in overrides, using first occurrence.", id);
-                                        }
+        // Override default capacities
+        if (brokerCapacity != null) {
+            // For checking for duplicate brokerIds
+            Set<Integer> overrideIds = new HashSet<>();
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java:380]: It looks like this was wrong already before. But maybe you can initialize this only later before the for loop on line 387? That way we won't do an unnecessary initialization.

**Review by scholzj — COMMENTED** [review]
Thanks for the PR. I left one nit. But I think it looks good otherwise.

**Review by kyguy — APPROVED** [review]
Looks @Xander-run, thanks for the fix! 

The unit test failures appear to be unrelated to the code changes, could you try running the flagged tests locally to confirm?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java (hunk 1 above)
Comment (Xander-run) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java:380]: sure thing

**Comment by Xander-run** [comment]
> The unit test failures appear to be unrelated to the code changes, could you try running the flagged tests locally to confirm?

I'm not quite sure what you mean by "flagged tests". Is it the [CruiseControl group](https://github.com/strimzi/strimzi-kafka-operator/blob/6d254718141c974f17f6fc4312cb029594b23e06/development-docs/TESTING.md?plain=1#L283) of the system test?

**Comment by scholzj** [comment]
@Xander-run There was a flaky test that failed originally. But I restarted the tests and it passed. You can just ignore it.

**Review by scholzj — APPROVED** [review]
LGTM. Thanks.

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks for the PR!

**Pull request merged**